GW2EIEvtcParser/EncounterLogic/Strikes/SotO/TempleOfFebe.cs

```diff
@@ -333,23 +522,214 @@ private static void AddHiddenWhileNotCasting(NPC target, ParsedEvtcLog log, Comb
+                        var oppositeRectangle = (RectangleDecoration)new RectangleDecoration(width, 100, lifespanIndicator, Colors.LightOrange, 0.2, oppositeAgentConnector).UsingRotationConnector(rotation);
+                        replay.AddDecorationWithGrowing(oppositeRectangle, growing);
+                        // Opposite Damage
+                        (long start, long end) lifespanDamageOpposite = (lifespanIndicator.end + 950, lifespanIndicator.end + 5850);
+                        (long start, long end) lifespanDamageOppositeCancelled = lifespanDamage;
+                        if (target.IsSpecies(TargetID.Cerus))
+                        {
+                            lifespanDamageOppositeCancelled = ComputeMechanicLifespanWithCancellationTime(log, lifespanDamageOpposite);
+                        }
+                        else
+                        {
+                            lifespanDamageOppositeCancelled = ComputeEmbodimentMechanicLifespanWithCancellationTime(log, lifespanDamageOpposite);
+                        }
+                        double millisecondsPerDegreeOpposite = (double)(lifespanDamageOpposite.end - lifespanDamageOpposite.start) / 360;
+                        double degreedRotatedOpposite = (lifespanDamageOppositeCancelled.end - lifespanDamageOppositeCancelled.start) / millisecondsPerDegreeOpposite;
+                        var rotation3 = new AngleConnector(facing, (float)degreedRotatedOpposite);
+                        var rectangle3 = (RectangleDecoration)new RectangleDecoration(width, 100, lifespanDamageOppositeCancelled, Colors.Red, 0.2, oppositeAgentConnector).UsingRotationConnector(rotation3);
+                        replay.Decorations.Add(rectangle3);
+                    }
+                }
+            }
+        }
+
+        private static (long start, long end) ComputeMechanicLifespanWithCancellationTime(ParsedEvtcLog log, (long start, long end) lifespan)
+        {
+            AbstractSingleActor cerus = log.FightData.GetMainTargets(log).Where(x => x.IsSpecies(TargetID.Cerus)).FirstOrDefault();
+            if (cerus != null)
+            {
+                var casts = cerus.GetCastEvents(log, log.FightData.FightStart, log.FightData.FightEnd).Where(x => x.SkillId == PetrifySkill).ToList();
+                foreach (AbstractCastEvent cast in casts)
+                {
+                    if (lifespan.start <= cast.Time && lifespan.end > cast.Time)
+                    {
+                        lifespan.end = Math.Min(lifespan.end, cast.Time);
+                    }
+                }
+            }
+            return lifespan;
+        }
+
+        private static (long start, long end) ComputeEmbodimentMechanicLifespanWithCancellationTime(ParsedEvtcLog log, (long start, long end) lifespan)
+        {
+            IReadOnlyList<PhaseData> phases = log.FightData.GetPhases(log).Where(x => !x.BreakbarPhase && x.CanBeSubPhase).ToList();
```

Comment:
This has a requirement on ComputePhases option being on, checking Cerus losing its invulnerability would be more robust